okio/src/main/java/okio/Buffer.java

```diff
@@ -775,6 +840,93 @@ public void clear() {
     return writeLong(reverseBytesLong(v));
   }
 
+  @Override public Buffer writeDecimalLong(long v) {
+    if (v == 0) {
+      // Both a shortcut and required since the following code can't handle zero.
+      return writeByte('0');
+    }
+
+    boolean negative = false;
+    if (v < 0) {
+      v = -v;
+      if (v < 0) { // Only true for Long.MIN_VALUE.
+        return writeUtf8("-9223372036854775808");
+      }
+      negative = true;
+    }
+
+    // Binary search for character width which favors matching lower numbers.
+    int width;
+    if (v < 100000000L) {
```

Comment:
(Perhaps with `_` chars afforded by Java 7, you could have something rather geometric?)